### PR TITLE
[PREVIEW] RPE-693 Refactor upload process

### PR DESCRIPTION
**Pull request by luigibk**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RPE-693

### Change description ###
Move pdf files check to doc error wrapper

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/ZipFileProcessorValidationTest.java [hunk 1]
```diff
@@ -24,27 +24,25 @@
 @RunWith(SpringRunner.class)
 public class ZipFileProcessorValidationTest {
 
-    private static final Processor processor = new Processor(
+    private static final EnvelopeProcessor envelopeProcessor = new EnvelopeProcessor(
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/ZipFileProcessorValidationTest.java]: class name is `ZipFileProcessorValidationTest`.
Seems like it's actually `EnvelopeProcessor` being tested here (or `Processor` previously).

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeControllerTest.java [hunk 2]
```diff
@@ -184,11 +184,10 @@ public void should_successfully_return_all_envelopes_with_processed_status_for_a
...
 
         assertThat(envelopeRepository.findAll())
-            .hasSize(2)
+            .hasSize(1)
             .extracting("zipFileName", "status")
             .containsExactlyInAnyOrder(
-                tuple("7_24-06-2018-00-00-00.zip", PROCESSED),
-                tuple("8_24-06-2018-00-00-00.zip", METADATA_FAILURE)
+                tuple("7_24-06-2018-00-00-00.zip", PROCESSED)
             );
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeControllerTest.java]: Not sure if this assertion belongs in a test for a controller.
Especially for tests for reading envelopes.

Perhaps something for another round of refactoring.

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/FileNameIrregularitiesException.java [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 import java.util.Collection;
 
-public class FileNameIrregularitiesException extends RuntimeException implements EnvelopeAwareThrowable {
+public class FileNameIrregularitiesException extends RuntimeException implements EventRelatedThrowable {
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/FileNameIrregularitiesException.java]: But we are aware of the envelope 🤔 . Perhaps we should still mark as failure and let the the retry limit allow re-parsing this envelope next time? Just need to be sure that counter is ++'ed

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeControllerTest.java [hunk 4]
```diff
@@ -184,11 +183,10 @@ public void should_successfully_return_all_envelopes_with_processed_status_for_a
             .andExpect(MockMvcResultMatchers.jsonPath("envelopes[0].id").exists());
 
         assertThat(envelopeRepository.findAll())
-            .hasSize(2)
+            .hasSize(1)
             .extracting("zipFileName", "status")
             .containsExactlyInAnyOrder(
-                tuple("7_24-06-2018-00-00-00.zip", PROCESSED),
-                tuple("8_24-06-2018-00-00-00.zip", METADATA_FAILURE)
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeControllerTest.java]: Looks like `METADATA_FAILURE` status can be remove as envelopes in such state will never be created after this change. (and such envelopes can be deleted from db).

Ok for next PR imho.

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/FileNameIrregularitiesException.java (hunk 3 above)
Comment (luigibk) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/FileNameIrregularitiesException.java]: The envelope is not created anymore when the files do not match the metadata. This is essentially treated as you'd treat a metadata invalid case as the envelope would be unusable anyway if created.

**Pull request merged**